src/components/Header/Header.js

```diff
@@ -88,7 +89,10 @@ class Header extends React.Component {
     const {
       patron,
       navData,
+      currentLocation = window.location || {},
```

Comment:
Have you tried running this branch in a web app? This fails:

![screen shot 2019-02-13 at 1 12 17 pm](https://user-images.githubusercontent.com/1280564/52733592-1fb17e00-2f91-11e9-919d-749c7c98772d.png)

Specifically, I pointed the homepage app to use this branch, compiled it, and then ran the app in production mode. You can't use `window` in the component unless if it's in the `componentDidMount` function since the server doesn't know what it is.

Reply:
Ahh I see..... I moved `window.location` because then I can pass currentLocation as a prop to Header.js. This way I can test Header.test.js based on different window.location since when we run test, `window.location` is undefined.

But now I get that `window.location` can't be called before componentDidMount

I have tried not to import `currentLocation` from `App.js` because it will require code changes for all the apps that import Header component. Maybe that is what we have to do?

I will investigate more solutions.

Reply:
OK~ I have modified the codebase. I followed the way @EdwinGuzman  tested and it successfully built on Hompage App locally.

I removed `window.location` out of `Header.js`, so now `currentLocation` as a prop will be solely passed from `app.js`.

Down side is that we have to modified the apps that import Header Component. But we can test Header Component's behaviors based on different locations now.

Reply:
I think it's sufficient to test the branch on one app instead of all of them since most run the same way.

You would have modified the app anyway in order to update the package version.

Note that:
* You _can_ just leave `window.location` in the header component package but you must include it on `componentDidMount`.
* If you leave the `currentLocation` to be updated by props, then this will **not** work in third-party apps like Encore, Locations, etc, because those sites do not pass `window.location` as a prop to the Header.

I would go with the first option of having `window.location` in componentDidMount and find other ways to test this. It's kind of unstable but it should work everywhere. I'd also recommend to test the branch in 1) a node app and 2) the Header app to test if it works on Encore, Locations, etc.

Reply:
How do you think if we implement the solution as below,

```
this.setLogOutLink(this.state.currentLocation);

setLogOutLink(location = window.location) {...}
```

and 

```
handleEncoreLoggedInTimer(currentLocation = window.location, currentTime) {...}
```

This way, if Header Component is on third party, we will still have a default `window.location`.

I think in this case, we can
1) make tests work.
2)sort the arguments for our functions. Most of them are from props.